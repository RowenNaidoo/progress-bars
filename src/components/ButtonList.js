import React from 'react'
import PropTypes from 'prop-types'
import Button from './Button'

const ButtonList = ({ buttons, onButtonClicked, limit }) => (
  buttons.map((button, index) =>
    <Button
      key={`button_${index}`}
      value={button}
      limit={limit}
      buttonClicked={onButtonClicked}
    />
  )
)

Button.propTypes = {
  buttons: PropTypes.arrayOf(Number),
  onButtonClicked: PropTypes.func,
  limit: PropTypes.number
}

export default ButtonList