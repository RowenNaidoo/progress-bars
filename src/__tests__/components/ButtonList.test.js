import React from 'react'
import { shallow } from 'enzyme'
import ButtonList from '../../components/ButtonList'


describe('#ButtonList', () => {

  test('renders learn react link', () => {
    const buttonClicked = jest.fn()
    const component = shallow(<ButtonList buttons={[7, 28, -7, -21]} onButtonClicked={buttonClicked} limit={150} />)
    expect(component).toMatchSnapshot()
  })

})