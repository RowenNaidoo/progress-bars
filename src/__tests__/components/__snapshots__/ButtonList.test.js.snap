// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ButtonList renders learn react link 1`] = `ShallowWrapper {}`;
